worlds/yoshisisland/Rules.py

```diff
@@ -229,77 +198,63 @@ def set_easy_rules(world, multiworld, player: int, boss_order, luigi_pieces):
-        set_rule(multiworld.get_location("More Monkey Madness: Stars", player), lambda state: True)
-        set_rule(multiworld.get_location("More Monkey Madness: Level Clear", player), lambda state: True)
-
-        set_rule(multiworld.get_location("The Impossible? Maze: Red Coins", player), lambda state: state.has_all({'Spring Ball', 'Large Spring Ball', 'Mole Tank Morph', 'Helicopter Morph', 'Flashing Eggs'}, player))
-        set_rule(multiworld.get_location("The Impossible? Maze: Flowers", player), lambda state: state.has_all({'Spring Ball', 'Large Spring Ball', 'Mole Tank Morph', 'Helicopter Morph'}, player))
-        set_rule(multiworld.get_location("The Impossible? Maze: Stars", player), lambda state: state.has_all({'Spring Ball', 'Large Spring Ball', 'Mole Tank Morph'}, player))
-        set_rule(multiworld.get_location("The Impossible? Maze: Level Clear", player), lambda state: state.has_all({'Spring Ball', 'Large Spring Ball', 'Mole Tank Morph', 'Helicopter Morph'}, player))
-
-        set_rule(multiworld.get_location("Kamek's Revenge: Red Coins", player), lambda state: state.has_all({'Key', 'Skis', 'Helicopter Morph', '! Switch'}, player) and logic.has_midring(state))
-        set_rule(multiworld.get_location("Kamek's Revenge: Flowers", player), lambda state: state.has_all({'Key', 'Skis', 'Helicopter Morph', '! Switch'}, player) and logic.has_midring(state))
-        set_rule(multiworld.get_location("Kamek's Revenge: Stars", player), lambda state: state.has_all({'! Switch'}, player) and logic.has_midring(state))
-        set_rule(multiworld.get_location("Kamek's Revenge: Level Clear", player), lambda state: state.has_all({'Key', 'Skis', '! Switch', 'Helicopter Morph'}, player))
-
-        set_rule(multiworld.get_location("Castles - Masterpiece Set: Red Coins", player), lambda state: (state.has('Egg Capacity Upgrade', player, 2) or logic.combat_item(state)) and state.has(('Large Spring Ball'), player))
-        set_rule(multiworld.get_location("Castles - Masterpiece Set: Flowers", player), lambda state: (state.has('Egg Capacity Upgrade', player, 2) or logic.combat_item(state)) and state.has(('Large Spring Ball'), player))
-        set_rule(multiworld.get_location("Castles - Masterpiece Set: Stars", player), lambda state: logic.has_midring(state) and state.has(('Large Spring Ball'), player))
-        set_rule(multiworld.get_location("Castles - Masterpiece Set: Level Clear", player), lambda state: (state.has('Egg Capacity Upgrade', player, 2) or logic.combat_item(state)) and state.has(('Large Spring Ball'), player))
+    set_easy_extra_rules(world, world.multiworld, world.player, world.boss_order, world.luigi_pieces)
+
+def set_easy_extra_rules(world, multiworld, player: int, boss_order, luigi_pieces):
+    if world.options.extras_enabled != 1:
+        return
+    logic = YoshiLogic(multiworld, player, boss_order, luigi_pieces, world)

```

Comment:
You could simple pass the YoshiLogic object as a parameter here, instead of passsing boss order and luigi_pieces then rebuilding it from scratch